### ISSUE TITLE
Questions about training results

### ISSUE THREAD

**Issue by Z-Zili** [JisongXie/StegaStamp_pytorch#8]
Did you achieve the effects of the author's paper? I use the model trained by your code to embed the information in the training set. The effect is as follows:
![im20_hidden](https://github.com/JisongXie/StegaStamp_pytorch/assets/87296854/e2d382a9-7e0e-4b95-aa99-0ec4a89347a6)
This image is from MIRFLICKR and I didn't change the parameters, batch_size=4,num_steps=144000. Do you know why? Or can you tell me any details that need attention in training?

**Comment by JisongXie** [comment]
yes, i can. however, I haven't worked on this for a long time.
It seems that you have trained a version of model, and when you inference it, the output is not as you expect.
I think you can check something as follow:
1. check the model training process, whether the training loss and validation loss decrease as expect, whether the training, validation and test acc are normal, and so on. In a word, ensure that the model is trained and saved correctly.
2. check the model inference process, including image preprocessing, model inference, and ouput postprocessing. From the image you show above, maybe your image values are [0, 1] or [0, 255], and your encoder's output, that is residual, is not in the same scale. Then when they sum together and are display, it looks strange. This is just my guess. maybe other reasons.
https://github.com/JisongXie/StegaStamp_pytorch/blob/184642e61d2fa95bcaf4131e64963202267c875d/encode_image.py#L74

**Comment by Z-Zili** [comment]
I'm glad you replied so quickly. Thank you for your answer, if I solve this problem later I will share my solution process.

**Comment by swyuan521** [comment]
Hello，The residual image I obtained after training the model can be decoded, but there are spots and wrinkles in the center of the image. Have you ever encountered this situation?
Thanks！
﻿@zziliang

**Comment by Z-Zili** [comment]
> Hello，The residual image I obtained after training the model can be decoded, but there are spots and wrinkles in the center of the image. Have you ever encountered this situation? Thanks！ ﻿@zziliang

I'm sorry, but I'm not in a similar situation.

**Comment by swyuan521** [comment]
Can you obtain a clear residual image and successfully decode it? Can you tell me where you made the modifications in the code?
Thank you very much!   @zziliang

**Comment by swyuan521** [comment]
Can we get in touch via email, WeChat, QQ, etc., please?
Thank you very much! @zziliang
